client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/embedded/RemoteQueryAccessTest.java

```diff
@@ -73,6 +77,65 @@ public void remoteQueries() {
       expectedIndexedQueries(2, 1);
    }
 
+   @DataProvider
+   static Object[][] batchSizes() {
+      return new Object[][] {
+            // Due to batch size being equal to 1 we need 3 queries when two elements are present. One for each value and
+            // a last for the empty result
+            {1, 1, 2, 1},
+            {1, 2, 4, 2},
+            {1, 10, 6, 3},
+            {1, -1, 6, 3},
+            // The batch size is exactly the same size as results so we need 2 queries, one with all values and
+            // another that is empty
+            {2, 1, 2, 1},
+            {2, 2, 2, 1},
+            {2, 10, 4, 2},
+            {2, -1, 4, 2},
+            // Remaining the batch size is larger than result size so we only need 1 query call per
+            {10, 1, 2, 1},
+            {10, 2, 2, 1},
+            {10, 10, 2, 1},
+            {10, -1, 2, 1},
+            {Integer.MAX_VALUE, 1, 2, 1},
+            {Integer.MAX_VALUE, 2, 2, 1},
+            {Integer.MAX_VALUE, 10, 2, 1},
+            {Integer.MAX_VALUE, -1, 2, 1},
+      };
+   }
+
+   @Test(dataProvider = "batchSizes")
+   public void remoteQueriesReactive(int batchSize, int maxReturn, int expectedIndexQueries, int expectedHybridQueries) {
+      RemoteCache<Object, Object> remoteCache = remoteCacheManager.getCache();
+      Query<Game> query = remoteCache.query(QUERY_TEXT);
+      query.maxResults(maxReturn);
+      List<Game> list = Flowable.fromPublisher(query.listAsync(batchSize))
+            .toList()
+            .blockingGet();
+      if (maxReturn == 1) {
+         assertThat(list).extracting("name")
+               .containsExactly("The Secret of Monkey Island");
```

Comment:
This is in the opposite order than expected in the test... it returns `Monkey Island 2: LeChuck's Revenge` first instead. Is this because the order is not guaranteed by the query? Maybe I should just change the assertion. @fax4ever please confirm.

Reply:
Exactly the order is not guaranteed if you don't put an `order by` clause to the Iclke text.

Reply:
K, just seemed odd to me that the test previously was asserting an exact ordering, but then when I limit the returned entry count it didn't have the same. I will change the test though to assume it always returns the second one though when a limit is applied.